packages/prisma/migrations/20250617123959_update_handle_routing_form_response_fields/migration.sql

```diff
@@ -0,0 +1,185 @@
+                        RAISE WARNING 'Failed to insert number value for field %', field_record->>'id';
+                    END;
+                END IF;
+            ELSIF field_type = 'select' THEN
+                -- Handle select values - can be either string or array
+                IF response_field->>'value' IS NOT NULL THEN
+                    BEGIN
+                        IF jsonb_typeof(response_field->'value') = 'array' THEN
+                            -- If it's an array, take the first element
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                (response_field->'value'->>0)
+                            );
+                        ELSIF jsonb_typeof(response_field->'value') = 'string' THEN
+                            -- If it's a string, use it directly
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                            VALUES (
+                                response_id,
+                                field_record->>'id',
+                                response_field->>'value'
+                            );
+                        END IF;
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert select value for field %', field_record->>'id';
+                    END;
+                END IF;
+            ELSE
+                -- Handle all other types as strings
+                -- Only insert if value is not null and is a valid string
+                IF response_field->>'value' IS NOT NULL AND jsonb_typeof(response_field->'value') = 'string' THEN
+                    BEGIN
+                        INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                        VALUES (
+                            response_id,
+                            field_record->>'id',
+                            response_field->>'value'
+                        );
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert string value for field %', field_record->>'id';
+                    END;
+                END IF;
+            END IF;
+        EXCEPTION WHEN OTHERS THEN
+            -- Log the error and continue processing other fields
+            RAISE WARNING 'Error processing field %', field_record->>'id';
+            CONTINUE;
+        END;
+    END LOOP;
+EXCEPTION WHEN OTHERS THEN
+    -- Log any unhandled errors and continue
+    RAISE WARNING 'Unhandled error in _process_routing_form_response_fields for response %', response_id;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Updated trigger function that uses the shared core function
+CREATE OR REPLACE FUNCTION handle_routing_form_response_fields()
```

Comment:
we used to have only `handle_routing_form_response_fields`, but now it's broken down to

- handle_routing_form_response_fields
- reprocess_routing_form_response_fields
- _process_routing_form_response_fields

Both `handle_routing_form_response_fields` and `reprocess_routing_form_response_fields` uses `_process_routing_form_response_fields` under the hood.

`handle_routing_form_response_fields` is used as a trigger (cannot be called by us manually).

We can call `reprocess_routing_form_response_fields` function manually to update the fields.